pkg/render/policyrecommendation.go

```diff
@@ -0,0 +1,329 @@
+	}
+
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(PolicyRecommendationNamespace, pr.cfg.PullSecrets...)...)...)
+
+	objs = append(objs,
+		pr.serviceAccount(),
+		pr.clusterRole(),
+		pr.clusterRoleBinding(),
+	)
+
+	// Deployment is for standalone or management cluster
+	if !pr.cfg.ManagedCluster {
+		objs = append(objs,
+			allowTigeraPolicyForPolicyRecommendation(pr.cfg),
+			pr.deployment(),
+		)
+	}
+
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(PolicyRecommendationNamespace, pr.cfg.ESSecrets...)...)...)
+
+	return objs, nil
+}
+
+func (pr *policyRecommendationComponent) Ready() bool {
+	return true
+}
+
+func (pr *policyRecommendationComponent) clusterRole() client.Object {
+	rules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"namespaces"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"projectcalico.org"},
+			Resources: []string{"licensekeys", "managedclusters"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"crd.projectcalico.org"},
+			Resources: []string{"licensekeys"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+		{
+			APIGroups: []string{"projectcalico.org"},
+			Resources: []string{
+				"tiers",
+				"policyrecommendationscopes",
+				"policyrecommendationscopes/status",
+				"stagednetworkpolicies",
+				"tier.stagednetworkpolicies",
+			},
+			Verbs: []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+		},
+	}
+
+	return &rbacv1.ClusterRole{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: PolicyRecommendationClusterRoleName,
+		},
+		Rules: rules,
+	}
+}
+
+func (pr *policyRecommendationComponent) clusterRoleBinding() client.Object {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1",
+			Kind:       "ClusterRoleBinding",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: PolicyRecommendationClusterRoleBindingName,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     PolicyRecommendationClusterRoleName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      PolicyRecommendationServiceAccountName,
+				Namespace: PolicyRecommendationNamespace,
+			},
+		},
+	}
+}
+
+func (pr *policyRecommendationComponent) controllerContainer() corev1.Container {
+	envs := []corev1.EnvVar{
+		{
+			Name:  "CLUSTER_NAME",
+			Value: pr.cfg.ESClusterConfig.ClusterName(),
+		},
+		{
+			Name:  "MULTI_CLUSTER_FORWARDING_CA",
+			Value: pr.cfg.TrustedBundle.MountPath(),
+		},
+		{
+			Name:  "FIPS_MODE_ENABLED",
+			Value: operatorv1.IsFIPSModeEnabledString(pr.cfg.Installation.FIPSMode),
+		},
+	}
+
+	sc := securitycontext.NewNonRootContext()
+
+	volumeMounts := []corev1.VolumeMount{}
```

Comment:
These three lines can be combined.

Reply:
Actually, PolicyRecommendationServerCertSecret is not used and the trustedbundle could as well be passed in-line, so this block can be removed.